apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProviderIT.java

```diff
@@ -900,126 +901,229 @@ public void searchForEobsByMissingPatient() {
   }
 
   /**
-   * Verifies that {@link
-   * gov.cms.bfd.server.war.r4.providers.ExplanationOfBenefitResourceProvider#findByPatient(ca.uhn.fhir.rest.param.ReferenceParam)}
-   * with <code>excludeSAMHSA=true</code> properly filters out SAMHSA-related claims.
+   * Load the SAMPLE_A resources and then tweak each of the claim types that support SAMHSA (all
+   * except PDE) to have a SAMHSA diagnosis code.
    *
-   * @throws FHIRException (indicates test failure)
+   * @return the beneficary record loaded by Sample A
    */
-  @Test
-  public void searchForEobsWithSamhsaFiltering() throws FHIRException {
+  private Beneficiary loadSampleAWithSamhsa() {
```

Comment:
It's easier to just pull data from the db and modify it slightly for the needs of the test, but I'm wondering if it would be worth just building claims as needed for the test in code, as it really only needs the values related to the SamhsaMatcher logic.  This would speed up these tests quite a bit not requiring a DB be set up for them.
For testing the endpoint, it would seem just a single (or two to test not set) tests would suffice, since testing if the `excludeSamhsa` parameter is set should be the same for all I think?

Reply:
What you described is basically the testing we did in the unit tests, where we added a lot more coverage and pulled from static files. For the integration tests we're going to err on the side of slightly more coverage and leave the four tests, since they each explicitly setup and test a scenario.